configuration/src/main/kotlin/com/malinskiy/marathon/config/serialization/ConfigurationFactory.kt

```diff
@@ -55,11 +55,23 @@ class ConfigurationFactory(
                 is VendorConfiguration.IOSConfiguration -> {
                     // Any relative path specified in Marathonfile should be resolved against the directory Marathonfile is in
                     val resolvedDerivedDataDir = marathonfileDir.resolve(configuration.vendorConfiguration.derivedDataDir)
-                    val finalXCTestRunPath = configuration.vendorConfiguration.xctestrunPath?.resolveAgainst(marathonfileDir)
-                        ?: fileListProvider
+                    val resolvedResultBundlePath = marathonfileDir.resolve(configuration.vendorConfiguration.xcResultBundlePath)
+
+                    // Adding support for Test Plan
+                    val testPlanName = configuration.vendorConfiguration.xcTestPlan
+
+                    var finalXCTestRunPath = if(!testPlanName.isNullOrEmpty()) {
+                        fileListProvider
                             .fileList(resolvedDerivedDataDir)
-                            .firstOrNull { it.extension == "xctestrun" }
-                        ?: throw ConfigurationException("Unable to find an xctestrun file in derived data folder")
+                            .firstOrNull { it.extension == "xctestrun" && it.name.contains("$testPlanName") } ?: throw ConfigurationException("Unable to find matching TestPlan. Please recheck if testplan is enabled")
```

Comment:
You can skip creating an additional string here:
`it.name.contains("$testPlanName")` -> `it.name.contains(testPlanName)`
It might also be a good idea to use startsWith or equals instead of contains. I'm not sure if there is a contract on the file name here